src/codec2_math.h

```diff
@@ -0,0 +1,50 @@
+#ifndef CODEC2_MATH_H
+#define CODEC2_MATH_H
+
+//==========================================================================
+// Name:            codec2_math.h
+//
+// Purpose:         A wrapper around architecture specific math libraries 
+//                  used on embedded devices to improve Codec2 performance.
+// Created:         May 15, 2022
+// Authors:         Mooneer Salem
+//
+// License:
+//
+//  This program is free software; you can redistribute it and/or modify
+//  it under the terms of the GNU Lesser General Public License version 2.1,
+//  as published by the Free Software Foundation.  This program is
+//  distributed in the hope that it will be useful, but WITHOUT ANY
+//  WARRANTY; without even the implied warranty of MERCHANTABILITY or
+//  FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public
+//  License for more details.
+//
+//  You should have received a copy of the GNU Lesser General Public License
+//  along with this program; if not, see <http://www.gnu.org/licenses/>.
+//
+//==========================================================================
+
+#if defined(__ARM_ARCH)
+#include "arm_math.h"
+
+#define codec2_dot_product_f32(left, right, len, result) (arm_dot_prod_f32(left, right, len, result))
+#define codec2_complex_dot_product_f32(left, right, len, resultReal, resultImag) (arm_cmplx_dot_prod_f32(left, right, len, resultReal, resultImag))
+
+#else
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif // __cplusplus
+
+// Your embedded project must define these two methods if it is not running on ARM.
+extern void codec2_dot_product_f32(float* left, float* right, size_t len, float* result);
+extern void codec2_complex_dot_product_f32(float* left, float* right, size_t len, float* resultReal, float* resultImag);
+
```

Comment:
Can we please make the variable names in the function headers a little more informative?  It would be nice to make it clear that the first two arguments are complex number vectors in floatReal, floatImag format (I think?).  Is `len` the number length in floats or complex numbers?  It would be nice to understand those sorts of details from the header alone.

Reply:
I updated the .h file to hopefully make the parameters more clear. Let me know if further clarifications are needed.

BTW I also added a `codec2_math_arm.c` so we don't need to use the preprocessor for the STM32 version of these.

Reply:
Thanks.  So the complex vectors have type float which is ambiguous.  Should we use the COMP struct or make it clear in the comments that they are floatReal,floatImag format?

Reply:
OK, updated comments accordingly.

Reply:
Taking another look at this one - I'm thinking we should use the COMP structure for the interface to make it consistent with other abstraction code like `codec2_fft.h` (and indeed the convention for the rest of libcodec2), and to make the interface clearer.  All that's required is to include `comp.h`.

I guess we also need to run all the stm32 ctests again.

Reply:
Haven't rerun the ctests yet (and probably won't have time to this weekend) but SM1000 does compile with the change I just pushed.